Add Trino 437 release notes (#20404)
trinodb/trino: docs/src/main/sphinx/release/release-437.md

# Release 437 (17 Jan 2024)

## General

* Add support for `char(n)` values in {func}`to_utf8`. ({issue}`20158`)
* Add support for `char(n)` values in {func}`lpad`. ({issue}`16907`)

> * Fix [table function-based](/functions/table) materialized views always being

Reviewer: I think this assumes too much .. maybe

Prevent permanent staleness of materialized views that use query pass-through table functions.

Wdyt about this suggestion instead @homar 

Also

Reviewer: @colebow ?

  stale. ({issue}`19904`)

## Delta Lake connector

> * Improve query performance for queries which don't involve table statistics. ({issue}`20054`)

Reviewer: ```suggestion
* Improve query performance for queries that don't use table statistics. ({issue}`20054`)
```

# Release 437 (17 Jan 2024)

## General

* Add support for `char(n)` values in {func}`to_utf8`. ({issue}`20158`)
* Add support for `char(n)` values in {func}`lpad`. ({issue}`16907`)
* {{breaking}} Replace the `exchange.compression-enabled` configuration property
  and `exchange_compression` session property with
  [the`exchange.compression-codec`and `exchange_compression_codec` properties](prop-exchange-compression-codec),
  respectively. ({issue}`20274`)
* {{breaking}} Replace the `spill-compression-enabled` configuration property 
  with [the `spill-compression-codec` property](prop-spill-compression-codec). ({issue}`20274`)
* {{breaking}} Remove the deprecated `experimental.spill-compression-enabled`
  configuration property. ({issue}`20274`)

> * Fix permanently stale materialized views when using pass-through table

Reviewer: What do you mean by "Fix MVs" ..

Author: I think "permanently stale MVs" is clear enough, no?

> * Fix permanently stale materialized views when using pass-through table

Reviewer 2: There's no such thing as a "pass-through" table function. Also, this issue has nothing to do with whether it's a a function that implements query pass-through or not -- all that matters is that the view references a table function.

  functions like `query` and `raw_query`. ({issue}`19904`)

## Delta Lake connector

* Improve query performance for queries that don't use table statistics. ({issue}`20054`)

## Hive connector

* Fix error when coercing union-typed data to a single type when reading Avro
  files. ({issue}`20310`)

## Iceberg connector

* Improve performance of queries with filters on `ROW` columns stored in Parquet
  files. ({issue}`17133`)

## SPI

> * Fix `ArrayIndexOutOfBoundsException` with RowBlockBuilder during output

Reviewer 2: Not an SPI change, but a bug fix. I'm not sure exactly when it manifests, though. @dain ?

  operations. ({issue}`20426`)

> * Fix adaptation of nullable function return to `BLOCK_BUILDER`. ({issue}`18456`)

Reviewer 2: This is not an SPI change. It's just a plain old bug fix. It should be something along the lines of:

* Fix failure when invoking functions that may return null values.

# Release 437 (17 Jan 2024)

## General

* Add support for `char(n)` values in {func}`to_utf8`. ({issue}`20158`)
* Add support for `char(n)` values in {func}`lpad`. ({issue}`16907`)
* {{breaking}} Replace the `exchange.compression-enabled` configuration property
  and `exchange_compression` session property with `exchange.compression-codec`

>   and `exchange_compression_codec`, respectively, which allow specifying `NONE`,

Reviewer: ```suggestion
  and `exchange_compression_codec`, respectively.
```

link to docs instead of having all the details here .. let me know if you need an anchor and I can add in https://github.com/trinodb/trino/pull/20458

  `LZ4`, or `ZSTD` as the compression codec for data exchanged between workers. ({issue}`20274`)
* {{breaking}} Replace the `spill-compression-enabled` configuration property 
  with `spill-compression-codec`, which allows specifying `NONE`, `LZ4`, or

>   `ZSTD` as the compression codec for data which is spilled to disk. ({issue}`20274`)

Reviewer: Same as prior item feedback
